security_groups.tf

```diff
@@ -281,6 +281,26 @@ resource "aws_security_group_rule" "hive_metastore_from_bgdc" {
   source_security_group_id = data.terraform_remote_state.adg.outputs.hive_metastore.security_group.id
 }
 
+resource "aws_security_group_rule" "bgdc_to_hive_metastore_v2" {
+  description              = "BGDC to Hive Metastore"
+  type                     = "ingress"
+  from_port                = 3306
+  to_port                  = 3306
+  protocol                 = "tcp"
+  source_security_group_id = aws_security_group.bgdc_common.id
+  security_group_id        = data.terraform_remote_state.internal_compute.outputs.hive_metastore_v2.security_group.id
+}
+
+resource "aws_security_group_rule" "hive_metastore_v2_from_bgdc" {
+  description              = "Hive Metastore from BGDC"
+  type                     = "egress"
+  from_port                = 3306
+  to_port                  = 3306
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.bgdc_common.id
+  source_security_group_id = data.terraform_remote_state.internal_compute.outputs.hive_metastore_v2.security_group.id
+}
+
```

Comment:
I was told that it might be best to keep all the SG rules for this with the provider of the service.
So inside internal compute. But regardless it should work anyway as the rules are correct

Reply:
We use this pattern of inserting SG rules remotely a lot as it follows the principle of layered, rather than circular dependencies between repos.